File: SocceropenApi/app/src/main/java/com/oscarmena/socceropenapi/VenuesAdapter.java
package com.oscarmena.socceropenapi;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.recyclerview.widget.RecyclerView;

import java.util.ArrayList;
import java.util.List;

public class VenuesAdapter extends RecyclerView.Adapter<VenuesAdapter.VenueViewHolder> {

    private List<VenuesResponse.Venue> venues = new ArrayList<>();

    @Override
    public VenueViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        View itemView = LayoutInflater.from(parent.getContext()).inflate(R.layout.activity_main, parent, false);
        return new VenueViewHolder(itemView);
    }

    @Override
    public void onBindViewHolder(VenueViewHolder holder, int position) {
        VenuesResponse.Venue venue = venues.get(position);
        holder.nameTextView.setText(venue.getName());
        holder.cityTextView.setText(venue.getCity());
        holder.addressTextView.setText(venue.getAddress());
        holder.capacityTextView.setText(String.valueOf(venue.getCapacity()));
    }

    @Override
    public int getItemCount() {
        return venues.size();
    }

    public void setVenues(List<VenuesResponse.Venue> venues) {
        this.venues = venues;
        notifyDataSetChanged();
    }

    public static class VenueViewHolder extends RecyclerView.ViewHolder {
        TextView nameTextView, cityTextView, addressTextView, capacityTextView;

        public VenueViewHolder(View itemView) {
            super(itemView);
            nameTextView = itemView.findViewById(R.id.tvName);
            cityTextView = itemView.findViewById(R.id.tvCity);
            addressTextView = itemView.findViewById(R.id.tvAddress);
            capacityTextView = itemView.findViewById(R.id.tvCapacity);
        }
    }
}

File: SocceropenApikotlin/app/src/main/java/com/oscarmena/socceropenapikotlin/VenuesAdapter.kt
package com.oscarmena.socceropenapikotlin

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView

class VenuesAdapter : RecyclerView.Adapter<VenuesAdapter.VenueViewHolder>() {

    private var venues: List<VenuesResponse.Venue> = listOf()

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): VenueViewHolder {
        val view = LayoutInflater.from(parent.context).inflate(R.layout.activity_main, parent, false)
        return VenueViewHolder(view)
    }

    override fun onBindViewHolder(holder: VenueViewHolder, position: Int) {
        val venue = venues[position]
        holder.nameTextView.text = venue.name
        holder.cityTextView.text = venue.cityName
        holder.addressTextView.text = venue.address
        holder.capacityTextView.text = venue.capacity?.toString()
    }

    override fun getItemCount(): Int = venues.size

    fun setVenues(venues: List<VenuesResponse.Venue>) {
        this.venues = venues
        notifyDataSetChanged()
    }

    class VenueViewHolder(view: View) : RecyclerView.ViewHolder(view) {
        val nameTextView: TextView = view.findViewById(R.id.tvName)
        val cityTextView: TextView = view.findViewById(R.id.tvCity)
        val addressTextView: TextView = view.findViewById(R.id.tvAddress)
        val capacityTextView: TextView = view.findViewById(R.id.tvCapacity)
    }
}